apigo/api/preview.go

```diff
@@ -63,15 +84,36 @@ func transform(resourceID string, rows []interface{}) (res []byte, err error) {
 
 // transform sends an HTTP request to the transformer service
 // using the PreviewRequest as JSON body. It returns the extrcted rows.
-func extract(req *PreviewRequest) (rows []interface{}, err error) {
-	jBody, _ := json.Marshal(req)
+func extract(preview *PreviewRequest, authorizationHeader string, idToken string) (rows []interface{}, err error) {
+	jBody, _ := json.Marshal(preview)
 	url := fmt.Sprintf("%s/extract", extractorURL)
-	resp, err := http.Post(url, "application/json", bytes.NewBuffer(jBody))
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jBody))
 	if err != nil {
 		return nil, err
 	}
+	if authorizationHeader != "" {
```

Comment:
what's the problem if you set an empty header ? except you'll get a 401 which is handled below

Reply:
No problemo, indeed